hledger-ui/Hledger/UI/TransactionScreen.hs

```diff
@@ -43,16 +45,20 @@ transactionScreen = TransactionScreen{
 tsInit :: Day -> Bool -> UIState -> UIState
 tsInit _d _reset ui@UIState{aopts=UIOpts{cliopts_=CliOpts{reportspec_=_rspec}}
                            ,ajournal=_j
-                           ,aScreen=TransactionScreen{}
+                           ,aScreen=s@TransactionScreen{tsTransaction=(_,t),tsTransactions=nts}
+                           ,aPrevScreens=prevscreens
                            } =
-  -- plog ("initialising TransactionScreen, value_ is "
-  --       -- ++ (pshow (Just (AtDefault Nothing)::Maybe ValuationType))
-  --       ++(pshow (value_ _ropts))  -- XXX calling value_ here causes plog to fail with: debug.log: openFile: resource busy (file is locked)
-  --       ++ "?"
-  --       ++" and first commodity is")
-  --       (acommodity$head$amounts$pamount$head$tpostings$snd$tsTransaction)
-  --      `seq`
-  ui
+    ui{aScreen=s{tsTransaction=(i',t'),tsTransactions=nts'}}
+  where
+    i' = maybe 0 (toInteger . (+1)) . elemIndex t' $ map snd nts'
+    -- If the previous screen was RegisterScreen, use the listed and selected items as
+    -- the transactions. Otherwise, use the provided transaction and list.
```

Comment:
This seem more complicated than before - now there's two ways to initialise a transaction screen, one of them unused.

But if we no longer need regenerateTransactions, that's great, and I'm ready to merge this.